lib/spack/spack/solver/runtimes.py

```diff
@@ -299,5 +299,8 @@ def _external_config_with_implicit_externals(configuration):
         if libc and libc not in seen:
             seen.add(libc)
             entry = {"spec": f"{libc}", "prefix": libc.external_path}
-            packages_yaml.setdefault(libc.name, {}).setdefault("externals", []).append(entry)
+            existing_entries = packages_yaml.setdefault(libc.name, {}).setdefault("externals", [])
+            if entry in existing_entries:
```

Comment:
There's already a `libc not in seen` branch. Why is that not sufficient? Nobody is manually configuring libc as an external.

Reply:
Then a better solution is to make it read-only...